frontend/public/components/image-stream.tsx

```diff
@@ -18,6 +18,46 @@ const menuActions = [
   ...common,
 ];
 
+// A valid specTag has a 'builder' tag. no 'hidden' tag, and exists in 'status.tags'.
+export const getBuilderSpecTagName = (imagestream) => {
+  let validSpecTag: any;
+
+  if (!imagestream.status) {
+    return null;
+  }
+
+  if (imagestream.spec && imagestream.spec.tags) {
+    validSpecTag = _.find(imagestream.spec.tags, (specTag: any) => {
+      let specTagTags: any = _.get(specTag, 'annotations.tags');
+      if (specTagTags) {
+        specTagTags = specTagTags.split(/\s*,\s*/);
+        if (_.includes(specTagTags, 'builder') && !_.includes(specTagTags, 'hidden')) {
+          return _.some(imagestream.status.tags, (statusTag: any) => {
+            return statusTag.tag === specTag.name;
+          });
+        }
```

Comment:
Let's avoid some of the nesting in this function to make it easier to understand.

```js
if (!_.includes(specTagTags, 'builder') || _.includes(specTagTags, 'hidden')) {
  return false;
}

return _.some(imagestream.status.tags, { tag: specTag.name });
```